wrapper/src/main/java/software/amazon/jdbc/util/monitoring/MonitorService.java

```diff
@@ -24,52 +24,64 @@
 public interface MonitorService {
   /**
    * Register a new monitor type with the monitor service. This method needs to be called before adding new types of
-   * monitors to the monitor service, so that the monitor service knows when a running monitor should be stopped.
+   * monitors to the monitor service, so that the monitor service knows when to dispose of a monitor.
    * Expected monitor types ("topology" and "customEndpoint") will be added automatically during driver initialization,
    * but this method can be called by users if they want to add a new monitor type.
    *
-   * @param monitorType       a String representing the monitor type, eg "customEndpoint".
-   * @param errorResponses    a Set defining actions to take if the monitor is in an error state.
-   * @param expirationTimeNs  how long a monitor should be stored before expiring, in nanoseconds. If the monitor is
-   *                          expired and shouldDisposeFunc returns `true`, the monitor will be stopped.
-   * @param shouldDisposeFunc a function defining whether an item should be stopped if expired. If `null` is passed, the
-   *                          monitor will always be stopped if the monitor is expired.
+   * @param monitorClass         the class of the monitor, eg `CustomEndpointMonitorImpl.class`.
+   * @param timeToLiveNanos      how long a monitor should be stored before being considered expired, in nanoseconds. If
+   *                             the monitor is expired and shouldDisposeFunc returns `true`, the monitor will be
+   *                             stopped.
+   * @param inactiveTimeoutNanos a duration in nanoseconds defining the maximum amount of time that a monitor should
```

Comment:
I like the idea of this argument which allows us to detect when monitors are frozen or stuck. However, I don't like that the difference between timeToLiveNanos and inactiveTimeoutNanos is hard to understand, even if we document it. To clarify, timeToLiveNanos represents a monitor's expiration, which is renewed if runIfAbsent is called but the monitor already exists. If the monitor has not been referenced via runIfAbsent in a long time it will be considered expired. inactiveTimeoutNanos represents a heartbeat threshold. If we haven't had a heartbeat within this threshold the monitor will be considered stuck.